modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/AsyncHttpClientClientGenerator.scala

```diff
@@ -767,26 +707,40 @@ object AsyncHttpClientClientGenerator {
                                                 AssignExpr.Operator.ASSIGN
                                               )
 
-                                            case RouteMeta.TextPlain =>
-                                              new AssignExpr(
-                                                new VariableDeclarationExpr(new VariableDeclarator(valueType, "result"), finalModifier),
-                                                new MethodCallExpr(new NameExpr("response"), "getResponseBody"),
-                                                AssignExpr.Operator.ASSIGN
-                                              )
-
-                                            case RouteMeta.OctetStream =>
-                                              // FIXME: need to standardize on a type for byte streams
-                                              new AssignExpr(
-                                                new VariableDeclarationExpr(new VariableDeclarator(valueType, "result"), finalModifier),
-                                                new MethodCallExpr(new NameExpr("response"), "getResponseBodyAsByteBuffer"),
-                                                AssignExpr.Operator.ASSIGN
+                                            case contentType =>
+                                              val bodyGetter = valueType match {
+                                                case _ if valueType.isNamed("InputStream") => "getResponseBodyAsStream"
+                                                case _ if valueType.isNamed("byte[]")      => "getResponseBodyAsBytes"
+                                                case _ if valueType.isNamed("ByteBuffer")  => "getResponseBodyAsByteBuffer"
+                                                case _ if valueType.isNamed("String")      => "getResponseBody"
+                                                case _ =>
+                                                  println(
+                                                    s"WARNING: Don't know how to handle response of type ${valueType.asString} for content type $contentType for operation ${operation.get.getOperationId}; falling back to String"
```

Comment:
Suggest threading through `Tracker` for `valueType` (in whatever form that takes). If that's not practical, at least bounding with `operation.history` would be better for context.

Reply:
Hmm, looking at where `valueType` comes from, not sure that's practical (ultimately it's from the `Responses[LA]` param passed through the term case class.  But yeah, `operation.history` at least is a good idea.